CRM/Contribute/Form/ContributionPage/Amount.php

```diff
@@ -87,22 +87,28 @@ public function buildQuickForm() {
 
     $this->addElement('checkbox', 'is_monetary', ts('Execute real-time monetary transactions'));
 
-    $paymentProcessor = CRM_Core_PseudoConstant::paymentProcessor();
-    $recurringPaymentProcessor = array();
-
-    if (!empty($paymentProcessor)) {
-      $paymentProcessorIds = implode(',', array_keys($paymentProcessor));
-      $query = "
-SELECT id
-  FROM civicrm_payment_processor
- WHERE id IN ({$paymentProcessorIds})
-   AND is_recur = 1";
-      $dao = CRM_Core_DAO::executeQuery($query);
-      while ($dao->fetch()) {
-        $recurringPaymentProcessor[] = $dao->id;
+    $paymentProcessors = CRM_Financial_BAO_PaymentProcessor::getAllPaymentProcessors('live');
+    $recurringPaymentProcessor = $futurePaymentProcessor = $paymentProcessor = array();
+
+    if (!empty($paymentProcessors)) {
+      foreach ($paymentProcessors as $id => $processor) {
+        if ($processor['name'] != 'pay_later') {
```

Comment:
NB I don't know you need to remove this

-      if ($contribution && $form->_values['is_recur'] && $contribution->contribution_recur_id
 -      ) {
 -        $paymentParams['contributionRecurID'] = $contribution->contribution_recur_id;
 -      }

There is some risk in that & while I'd like to see it go I think you are taking more risk on your change than you need to by removing it. Maybe you should alter it to not overwrite inappropriately instead?